src/components/WorkspaceDeploymentDialog.tsx

```diff
@@ -0,0 +1,274 @@
+import { Icon } from '@iconify/react';
+import { EditorDialog } from '@kinvolk/headlamp-plugin/lib/CommonComponents';
+import {
+  Alert,
+  Box,
+  Button,
+  Dialog,
+  DialogActions,
+  DialogContent,
+  DialogTitle,
+  Divider,
+  Paper,
+  Stack,
+  Typography,
+} from '@mui/material';
+import yaml from 'js-yaml';
+import React, { useRef, useState } from 'react';
+import NodeSelector from './NodeSelector';
+
+interface PresetModel {
+  name: string;
+  version: string;
+  company: {
+    name: string;
+    url: string;
+  };
+  supportsTools: boolean;
+  logoImageId: string;
+  description: string;
+  instanceType: string;
+}
+
+interface WorkspaceDeploymentDialogProps {
+  open: boolean;
+  onClose: () => void;
+  model: PresetModel | null;
+  onDeploy?: (_yamlContent: string) => void;
+}
+
+const WorkspaceDeploymentDialog: React.FC<WorkspaceDeploymentDialogProps> = ({
+  open,
+  onClose,
+  model,
+  onDeploy: _onDeploy,
+}) => {
+  const [selectedNodes, setSelectedNodes] = useState<string[]>([]);
+  const [labelSelector, setLabelSelector] = useState<string>('');
+  const [editorDialogOpen, setEditorDialogOpen] = useState(false);
+  const [_editorValue, setEditorValue] = useState('');
+  const [requiredNodes, setRequiredNodes] = useState<number | ''>('');
+
+  const itemRef = useRef({});
+
+  const generateWorkspaceYAML = (model: PresetModel, preferredNodes: string[]): string => {
+    const modelNameCheck = model.name.toLowerCase().replace(/[^a-z0-9-]/g, '-');
+    const isLlama = model.name.toLowerCase().includes('llama');
+
+    // Use different label selector based on whether nodes are selected
+    const labelSelector =
+      preferredNodes.length > 0
+        ? 'node.kubernetes.io/instance-type: Standard_NC80adis_H100_v5'
```

Comment:
Can we not hardcode the instance type and instead, get it from the preferred nodes user selects?